File: src/Tests.cs/BifrostTests.cs
using System;
using FluentAssertions;
using NUnit.Framework;
using TLK.LoveDeathFSharp.Kernel;

namespace TLK.LoveDeathFSharp.Tests
{
    internal class BifrostTests
    {
        [Test]
        public void HofundShouldOpenBifrost()
        {
            // arrange
            var hofund = CreateHofund();

            // act
            Action act = () => new Bifrost(hofund);

            // assert
            act.Should().NotThrow();
        }

        [Test]
        public void InfinityGauntletShouldNotOpenBifrost()
        {
            // arrange
            var infinityGauntlet = CreateInfinityGauntlet();

            // act
            Action act = () => new Bifrost(infinityGauntlet);

            // assert
            act.Should().Throw<Exception>();
        }

        private static Hofund CreateHofund() => new Hofund();
        private static InfinityGauntlet CreateInfinityGauntlet() => new InfinityGauntlet();
    }
}

File: src/Tests.fs/BifrostTests.fs
module TLK.PortalsAndHeroes.Tests.BifrostTests

open NUnit.Framework
open TLK.PortalsAndHeroes.Kernel

let testDelegate func = new TestDelegate(func)
let shouldNotThrow = testDelegate >> Assert.DoesNotThrow
let shouldThrow action = Assert.That(testDelegate action, Throws.Exception)

let hofund = new Hofund()
let infinityGauntlet = new InfinityGauntlet()
let openBifrost key () = new Bifrost(key) |> ignore

[<Test>]
let ``Hofund should open Bifrost`` () =
    hofund |> openBifrost |> shouldNotThrow

[<Test>]
let ``Infinity Gauntlet should not open Bifrost`` () =
    infinityGauntlet |> openBifrost |> shouldThrow
